smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -59,7 +66,74 @@ public KafkaSource(Vertx vertx, KafkaConnectorIncomingConfiguration config) {
+                    final long consumerReEnableWaitTime = Long.parseLong(
+                            kafkaConfiguration.getOrDefault(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, "300000"))
+                            + (kafkaConfiguration.get(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG) == null ? 0L
+                                    : Long.parseLong(
+                                            kafkaConfiguration.getOrDefault(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG,
+                                                    "10000")));
+
+                    // We will retry the re-balance consumer listener on failure using an exponential backoff until
+                    // we can allow the kafka consumer to do it on its own. We do this because by default it would take
+                    // 5 minutes for kafka to do this which is too long. With defaults consumerReEnableWaitTime would be
+                    // 500000 millis. We also can't simply retry indefinitely because once the consumer has been paused
+                    // for consumerReEnableWaitTime kafka will force a re-balance once resumed.
+                    // We are doing retries using the time intervals 2s, 4s, 8s, 10s, 10s, 10s, 10s...
+                    // The following formula will give us a reasonable number for retry attempt that is just greater
+                    // than consumerReEnableWaitTime
+                    final long consumerReEnableRetryMaxAttempts = 1
+                            + Math.max(0, 3 + (consumerReEnableWaitTime - 14_000) / 10_000);
+
+                    kafkaConsumer.partitionsAssignedHandler(set -> {
+                        kafkaConsumer.pause();
+                        log.executingConsumerAssignedRebalanceListener(group);
+                        listener.onPartitionsAssigned(kafkaConsumer, set)
+                                .onFailure().invoke(t -> log.unableToExecuteConsumerAssignedRebalanceListener(group, t))
+                                .onFailure().retry().withBackOff(Duration.ofSeconds(1), Duration.ofSeconds(10)).withJitter(0.0)
+                                .atMost(consumerReEnableRetryMaxAttempts)
+                                .onFailure().invoke(t -> {
+                                    log.reEnablingConsumerforGroup(group);
+                                    kafkaConsumer.resume();
+                                })
+                                .onItem().invoke(a -> kafkaConsumer.resume())
+                                .subscribe()
+                                .with(a -> log.executedConsumerAssignedRebalanceListener(group));
+                    });
+
+                    kafkaConsumer.partitionsRevokedHandler(set -> {
+                        log.executingConsumerRevokedRebalanceListener(group);
+                        listener.onPartitionsRevoked(kafkaConsumer, set)
+                                .onFailure().invoke(t -> log.unableToExecuteConsumerRevokedRebalanceListener(group, t))
```

Comment:
I would recommend moving this to the subscribe.

Reply:
done